warehouse/query-core/src/main/java/datawave/query/jexl/functions/TermFrequencyAggregatorFactory.java

```diff
@@ -0,0 +1,9 @@
+package datawave.query.jexl.functions;
+
+import datawave.query.predicate.EventDataQueryFilter;
+
+import java.util.Set;
+
+public interface TermFrequencyAggregatorFactory {
+    TermFrequencyAggregator create(Set<String> fieldsToAggregate, EventDataQueryFilter attrFilter, int maxNextCount);
```

Comment:
Instead of passing the Iterator Building Visitor to the iterator builders, and calling create within the iterator builders, why not create the the TermFrequencyAggregator in the iterator building visitor and pass that into the iterator builder.  It feels odd to be passing the iterator building visitor around to the iterator builders.  In that case, do we even need the interface?

Reply:
I can refactor buildExceededFromTermFrequency() to call some new method in the IteratorBuildingVisitor which can be overridden in the TLD version to generate the correct TermFrequencyAggregator.

Reply:
That sounds good to me!